me.id.webverify/webverifylib/src/main/java/me/id/webverifylib/WebViewActivity.java

```diff
@@ -18,23 +29,115 @@
   public static final String EXTRA_URL = "url";
 
   protected WebView webView;
+  protected Toolbar toolbar;
+  protected WebView progressBarWebView;
+  protected Button networkErrorButton;
+  protected View internetConnectionErrorView;
   protected IDmeScope scope;
   protected String url;
 
   @Override
-  @SuppressLint("SetJavaScriptEnabled")
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_web_view);
 
     scope = getScopeFromKey(getIntent().getStringExtra(EXTRA_SCOPE_ID));
     url = getIntent().getStringExtra(EXTRA_URL);
 
+    initializeNoInternetConnectionView();
+    initializeProgressBar();
+    initializeToolbar();
+    initializeWebView();
+  }
+
+  @SuppressLint("SetJavaScriptEnabled")
+  private void initializeWebView() {
     webView = (WebView) findViewById(R.id.webView);
+    webView.getSettings().setAppCacheEnabled(false);
+    webView.getSettings().setCacheMode(WebSettings.LOAD_NO_CACHE);
+    webView.getSettings().setJavaScriptEnabled(true);
     clearWebViewCacheAndHistory();
     webView.setWebViewClient(getWebClient(scope));
     webView.loadUrl(url);
-    webView.getSettings().setJavaScriptEnabled(true);
+  }
+
+  private void initializeToolbar() {
+    toolbar = (Toolbar) findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+    //noinspection ConstantConditions
+    getSupportActionBar().setTitle("");
+    supportInvalidateOptionsMenu();
+  }
+
+  private void hideProgressBar(int startDelayInSeconds) {
+    if (progressBarWebView != null) {
+      progressBarWebView.animate()
+          .alpha(0.0f)
+          .setDuration(1000)
+          .setStartDelay(startDelayInSeconds * 1000)
+          .setListener(new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+              super.onAnimationEnd(animation);
+              progressBarWebView.setVisibility(View.GONE);
+            }
+          });
+    }
+  }
+
+  private void hideInternetConnectionErrorView(int startDelayInSeconds) {
+    if (progressBarWebView != null) {
+      internetConnectionErrorView.animate()
+          .alpha(0.0f)
+          .setDuration(500)
+          .setStartDelay(startDelayInSeconds * 1000)
```

Comment:
is `1000` ok here? in the previous function you are using the same value passed to `duration`